packages/transport/src/chrome-runtime/adapter.ts

```diff
@@ -0,0 +1,206 @@
+   * router.
+   *
+   * Connectrpc enforces a separation between client-side ContextValues and
+   * server-side ContextValues, even though they use the same types. You can't
+   * simply provide contextValues to a client's CallOptions - those are only for
+   * Interceptors running in your client-side Transport. They won't be forwarded
+   * to the server.
+   *
+   * Server framework adapters are responsible for applying context values to
+   * the implemenation. Normally, this would be a simple parameter in a call to
+   * the function connectNodeAdapter.  But, we're not using node.
+   *
+   * And unfortunately, createUniversalHandlerClient which createRouterTransport
+   * uses internally doesn't pass a context parameter at all when it calls a
+   * handler - meaning every HandlerContext has no contextValues. So the only
+   * way to get context to the implementation is to either createHandlerContext
+   * yourself and call the Impl directly, or createContextValues yourself and
+   * call the router directly.
+   *
+   * I've chosen to reimplement the simple createRouterTransport, but wrap the
+   * router's handlers, inserting contextValues into the requests made by the
+   * UniversalClient. At request time, this kind of context injector only has
+   * access to data present in the UniversalServerRequest.
+   *
+   * It is probably ideal to write a real adapter, but in then we must deal more
+   * directly with connectrpc's expectations of operating as an http server. It
+   * seems that a router transport is the endorsed way to route local services.
+   *
+   * See https://github.com/connectrpc/connect-es/blob/main/packages/connect-node/src/node-universal-handler.ts
+   */
+  const injectRequestContext = createUniversalHandlerClient(
+    router.handlers.map((handler: UniversalHandler): UniversalHandler => {
+      // disaggregate the handler into actual function and keyed attributes
+      const handlerFn = handler as UniversalHandlerFn;
+      const handlerMeta = Object.fromEntries(
+        Object.entries(handler) as [keyof UniversalHandler, unknown][],
+      ) as { [k in keyof UniversalHandler]: UniversalHandler[k] };
+      // wrap the handler function to generate and apply context
+      const wrappedFn: UniversalHandlerFn = req =>
+        opt.createRequestContext(req).then(handlerFn, error => {
+          // it's convenient to log and rethrow here
+          console.warn('Handler Error', handler.name, error);
+          throw error;
+        });
+      // replace attributes onto the wrapped handler
+      return Object.assign(wrappedFn, handlerMeta);
+    }),
+  );
+
+  // TODO: alternatively, we could have the channelClient provide a requestPath
+  const I_MethodType = new Map<string, MethodType>(
+    router.handlers.map(({ method, service }) => [
+      method.I.typeName,
+      { ...method, service: { typeName: service.typeName } },
+    ]),
+  );
+
+  // TODO: interceptors?
```

Comment:
What are interceptors? Can you make a more descriptive TODO if it's for later PRs?